src/form/wr.jl

```diff
@@ -736,3 +737,97 @@ function constraint_power_magnitude_link_on_off(pm::QCWRPowerModel, n::Int, i, a
 
     @constraint(pm.model, cm == (g^2 + b^2)*(w_fr/tm + w_to - 2*(tr*wr + ti*wi)/tm) - c*q_fr - ((c/2)/tm)^2*w_fr)
 end
+
+
+""
+@compat abstract type QCWRTriForm <: QCWRForm end
+
+""
+const QCWRTriPowerModel = GenericPowerModel{QCWRTriForm}
+
+"default QC trilinear model constructor"
+function QCWRTriPowerModel(data::Dict{String,Any}; kwargs...)
+    return GenericPowerModel(data, QCWRTriForm; kwargs...)
+end
+
+""
+function variable_voltage_magnitude_product{T <: QCWRTriForm}(pm::GenericPowerModel{T}, n::Int=pm.cnw)
+    # do nothing - no lifted variables required for voltage variable product
+end 
+
+"creates lambda variables for convex combination model" 
+function variable_multipliers{T <: QCWRTriForm}(pm::GenericPowerModel{T}, n::Int=pm.cnw)
+    buspairs = pm.ref[:nw][n][:buspairs]
+    pm.var[:nw][n][:lambdar] = @variable(pm.model, 
+        [bp in keys(pm.ref[:nw][n][:buspairs]), i=1:8], basename="$(n)_lambda",
+        lowerbound = 0, upperbound = 1)
+    pm.var[:nw][n][:lambdai] = @variable(pm.model, 
+        [bp in keys(pm.ref[:nw][n][:buspairs]), i=1:8], basename="$(n)_lambda",
+        lowerbound = 0, upperbound = 1)
+end
+
+""
+function variable_voltage(pm::QCWRTriPowerModel, n::Int=pm.cnw; kwargs...)
+    variable_voltage_angle(pm, n; kwargs...)
+    variable_voltage_magnitude(pm, n; kwargs...)
+
+    variable_voltage_magnitude_sqr(pm, n; kwargs...)
+    variable_voltage_product(pm, n; kwargs...)
+
+    variable_voltage_angle_difference(pm, n; kwargs...)
+    variable_voltage_magnitude_product(pm, n; kwargs...)
+    variable_multipliers(pm, n; kwargs...)
+    variable_cosine(pm, n; kwargs...)
+    variable_sine(pm, n; kwargs...)
+    variable_current_magnitude_sqr(pm, n; kwargs...)
+end
+
+""
+function constraint_voltage(pm::QCWRTriPowerModel, n::Int)
+    v = pm.var[:nw][n][:vm]
+    t = pm.var[:nw][n][:va]
+
+    td = pm.var[:nw][n][:td]
+    si = pm.var[:nw][n][:si]
+    cs = pm.var[:nw][n][:cs]
+
+    w = pm.var[:nw][n][:w]
+    wr = pm.var[:nw][n][:wr]
+    lambdar = pm.var[:nw][n][:lambdar]
```

Comment:
might consider calling these `lambda_wr` and `lambda_wi`, to connect them more explicitly to the `wr` and `wi ` variables